buildSrc/src/main/java/gradlebuild/JniNokeePlugin.java

```diff
@@ -0,0 +1,213 @@
+import dev.nokee.language.cpp.CppSourceSet;
+import dev.nokee.platform.jni.JavaNativeInterfaceLibrary;
+import dev.nokee.runtime.nativebase.TargetMachine;
+import dev.nokee.runtime.nativebase.TargetMachineFactory;
+import groovy.util.Node;
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.plugins.ExtensionAware;
+import org.gradle.api.plugins.JavaPluginConvention;
+import org.gradle.api.provider.SetProperty;
+import org.gradle.api.publish.PublishingExtension;
+import org.gradle.api.publish.maven.MavenPublication;
+import org.gradle.api.tasks.SourceSet;
+import org.gradle.api.tasks.TaskContainer;
+import org.gradle.api.tasks.TaskProvider;
+import org.gradle.api.tasks.compile.JavaCompile;
+import org.gradle.language.cpp.tasks.CppCompile;
+import org.gradle.model.Mutate;
+import org.gradle.model.RuleSource;
+import org.gradle.nativeplatform.platform.OperatingSystem;
+import org.gradle.nativeplatform.platform.internal.DefaultNativePlatform;
+import org.gradle.nativeplatform.tasks.LinkSharedLibrary;
+import org.gradle.nativeplatform.toolchain.Clang;
+import org.gradle.nativeplatform.toolchain.Gcc;
+import org.gradle.nativeplatform.toolchain.NativeToolChainRegistry;
+import org.gradle.nativeplatform.toolchain.VisualCpp;
+
+import java.io.File;
+import java.util.Set;
+
+@SuppressWarnings("UnstableApiUsage")
+public abstract class JniNokeePlugin implements Plugin<Project> {
+
+    @Override
+    public void apply(Project project) {
+        project.getPluginManager().apply(NativePlatformComponentPlugin.class);
+        VariantsExtension variants = project.getExtensions().getByType(VariantsExtension.class);
+        project.getPluginManager().apply("dev.nokee.jni-library");
+        project.getPluginManager().apply(NativeToolChainRules.class);
+
+        configureCppTasks(project);
+        configureMainLibrary(project);
+        configureVariants(project);
+        addComponentSourcesSetsToProjectSourceSet(project.getTasks(), project.getExtensions().getByType(JavaNativeInterfaceLibrary.class));
+
+        configureNativeVersionGeneration(project);
+
+        configurePomOfMainJar(project, variants);
+    }
+
+    private void configureVariants(Project project) {
+        JavaNativeInterfaceLibrary library = project.getExtensions().getByType(JavaNativeInterfaceLibrary.class);
+        VariantsExtension variants = project.getExtensions().getByType(VariantsExtension.class);
+        variants.getVariantNames().set(library.getVariants().flatMap(it -> {
+            // Only depend on variants which can be built on the current machine
+            boolean onlyLocalVariants = project.getProviders().gradleProperty("onlyLocalVariants").forUseAtConfigurationTime().isPresent();
+            if (onlyLocalVariants && !it.getSharedLibrary().isBuildable()) {
+                return ImmutableList.of();
+            } else {
+                return ImmutableList.of(toVariantName(it.getTargetMachine()));
+            }
+        }));
```

Comment:
What does it mean to set the "variant names" here? This disables certain variants from running?